packages/core/src/RenderingEngine/StackViewport.ts

```diff
@@ -816,13 +816,38 @@ class StackViewport extends Viewport implements IStackViewport {
     this.setVOIGPU(voiRange, suppressEvents);
   }
 
-  private setRotation(rotationCache: number, rotation: number): void {
+  protected flip({ flipHorizontal, flipVertical }: FlipDirection): void {
+    super.flip({ flipHorizontal, flipVertical });
+
+    //
+    // This viewport always applies (a clock-wise) rotation last. For a horizontal
+    // flip this means to negate the rotation because using matrix notation the
+    // following is true...
+    // [-1 0][cos  sin]  =  [cos -sin][-1 0]
+    // [ 0 1][-sin cos]     [sin  cos][ 0 1]
+    //
+    // A similar thing holds true for vertical flips below.
+    //
+    if (flipHorizontal) {
+      this.rotationCache = -this.rotationCache;
+    }
+
+    if (flipVertical) {
+      this.rotationCache = -this.rotationCache;
+    }
+  }
```

Comment:
This is the only tricky part ....
So if it is flipped, and someone do the `getProperties` the rotation will be negative?

Reply:
That is correct. However we could make it positive if that is desirable.